packages/twenty-server/src/engine/metadata-modules/search/search.service.ts

```diff
@@ -0,0 +1,177 @@
+      createdObjectMetadata.workspaceId,
+      [
+        {
+          name: computeTableName(
+            createdObjectMetadata.nameSingular,
+            createdObjectMetadata.isCustom,
+          ),
+          action: WorkspaceMigrationTableActionType.ALTER,
+          columns: this.tsVectorColumnActionFactory.handleCreateAction({
+            ...searchVectorFieldMetadata,
+            defaultValue: undefined,
+            generatedType: 'STORED',
+            asExpression: getTsVectorColumnExpressionFromFields([
+              {
+                type: searchableFieldForCustomObject.type as SearchableFieldType,
+                name: searchableFieldForCustomObject.name,
+              },
+            ]),
+            options: undefined,
+          } as FieldMetadataInterface<FieldMetadataType.TS_VECTOR>),
+        },
+      ],
+    );
+
+    await this.indexMetadataService.createIndex(
+      objectMetadataInput.workspaceId,
+      createdObjectMetadata,
+      [searchVectorFieldMetadata],
+      false,
+      IndexType.GIN,
+    );
+  }
+
+  public async updateSearchVector(
+    objectMetadataId: string,
+    labelIdentifierFieldMetadataId: string,
+    workspaceId: string,
+  ) {
+    const newLabelIdentifierField =
+      await this.fieldMetadataRepository.findOneByOrFail({
+        id: labelIdentifierFieldMetadataId,
+      });
+
+    if (!isSearchableFieldType(newLabelIdentifierField.type)) {
+      return;
+    }
+
+    const objectMetadata = await this.objectMetadataRepository.findOneByOrFail({
+      id: objectMetadataId,
+    });
+
+    const existingSearchVectorFieldMetadata =
+      await this.fieldMetadataRepository.findOneByOrFail({
+        name: SEARCH_VECTOR_FIELD.name,
+        objectMetadataId,
+      });
+
+    await this.workspaceMigrationService.createCustomMigration(
+      generateMigrationName(`update-${objectMetadata.nameSingular}`),
+      workspaceId,
+      [
+        {
+          name: computeTableName(
+            objectMetadata.nameSingular,
+            objectMetadata.isCustom,
+          ),
+          action: WorkspaceMigrationTableActionType.ALTER,
+          columns: this.workspaceMigrationFactory.createColumnActions(
+            WorkspaceMigrationColumnActionType.ALTER,
+            existingSearchVectorFieldMetadata,
+            {
+              ...existingSearchVectorFieldMetadata,
+              asExpression: getTsVectorColumnExpressionFromFields([
+                {
+                  type: newLabelIdentifierField.type as SearchableFieldType,
+                  name: newLabelIdentifierField.name,
+                },
+              ]),
+              generatedType: 'STORED', // Not stored on fieldMetadata
+              options: undefined,
+            },
+          ),
+        },
+      ],
+    );
+
+    // index needs to be recreated as typeorm deletes then recreates searchVector column at alter
+    await this.indexMetadataService.createIndex(
```

Comment:
@Weiko what happens in that typeOrm deletes then recreates searchVector column during update (cf PostGresQuerryRunner - changeColumn method - L751:
```
if (oldColumn.type !== newColumn.type ||
            oldColumn.length !== newColumn.length ||
            newColumn.isArray !== oldColumn.isArray ||
            (!oldColumn.generatedType &&
                newColumn.generatedType === "STORED") ||
            (oldColumn.asExpression !== newColumn.asExpression &&
                newColumn.generatedType === "STORED")) {
            // To avoid data conversion, we just recreate column
            await this.dropColumn(table, oldColumn);
            await this.addColumn(table, newColumn);
            // update cloned table
            clonedTable = table.clone();
        }
```)

So the index on the table is deleted as well, but the index metadata is untouched, creating a temporary inconsistancy. 
When I recreate the index right after, nothing new is saved on the index metadata table as the same information are to be saved, and the index is re-created on the table, so we are back to a consistent state.

Does that sound too fragile? This could also be a potential candidate for another patch

Reply:
Interesting!